### PR TITLE
fix: Authentication for plugins. Fixes #8144

### LINKED ISSUE #8144: Require authentication for plugins
Currently plugins do not require authentication. As pod listen permissively by default (which is a dangerous security posture that Kubernetes takes), this allows any other pod in the cluster to make inbound requests to it. We should require authetication.

### PR DESCRIPTION
**Pull request by alexec**
Signed-off-by: Alex Collins <alex_collins@intuit.com>

Fixes #8144

### CODE REVIEW

File: workflow/controller/agent.go [hunk 1]
```diff
@@ -111,6 +112,48 @@ func (woc *wfOperationCtx) createAgentPod(ctx context.Context) (*apiv1.Pod, erro
...
+		},
+		Resources: apiv1.ResourceRequirements{
+			Requests: map[apiv1.ResourceName]resource.Quantity{
+				"cpu":    resource.MustParse("10m"),
+				"memory": resource.MustParse("64M"),
+			},
+			Limits: map[apiv1.ResourceName]resource.Quantity{
+				"cpu":    resource.MustParse(env.LookupEnvStringOr("ARGO_AGENT_CPU_LIMIT", "100m")),
+				"memory": resource.MustParse(env.LookupEnvStringOr("ARGO_AGENT_MEMORY_LIMIT", "256M")),
+			},
```
Comment (terrytangyuan) [inline workflow/controller/agent.go:137]: Consider parameterizing these?

Comment (alexec) [reply workflow/controller/agent.go:137]: I'm not sure about this. The agent's memory and CPU usage should be well-defined. It actually (and rarely) should not be different of different users set-up, as we use a watch not informer, and fixed size worked pool.

Configuation would hint at bugs.

File: cmd/argoexec/commands/agent.go [hunk 2]
```diff
@@ -20,9 +22,59 @@ import (
...
+				if err := os.WriteFile(filename, []byte(token), 0o440); err != nil {
+					log.Fatal(err)
+				}
+			}
+		},
+	}
+}
+
+func tokenFilename(name string) string {
+	return filepath.Join("/var/run/argo", name, "token")
```
[the remark this answers is not in the document]
Comment (alexec) [reply cmd/argoexec/commands/agent.go:56]: good idea

**Review by terrytangyuan — COMMENTED** [review]
LGTM.

File: cmd/argoexec/commands/agent.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (alexec) [reply cmd/argoexec/commands/agent.go]: done

**Review by terrytangyuan — APPROVED** [review]

**Pull request merged**